### ISSUE TITLE
Miner did not detect diamond ore

### ISSUE THREAD

**Issue by Magleose** [TangoTek/TekTopia-Community#227]
**Describe the bug**
I created a new mineshaft, and walked away for a day cycle, came back to check on the progress, and the miner mined right past diamond ore in the area it is supposed to be detecting ore in

**To Reproduce**
1. I don't know if you can

**Expected behavior**
I expect the miner to gather ore in all the areas it can scan

**Screenshots**
![2019-05-29_22 34 44](https://user-images.githubusercontent.com/37353615/58719394-83738180-839d-11e9-8ff6-b7afbcc7bb51.png)

** Server/Location **
 Main(Survival) 2
1357 / 70 / 5261

**Additional context**
I am not sure if you can reproduce, but I left the diamonds where they were in my mine in case you want to observer this. You can find the mine at 1397 / 12 / 5256

The miners have picked up other diamonds, just for some reason left these ones. I have no druid at the point of submitting this report

**Comment by JohnyDL** [comment]
They only check the blocks immediately adjasent to the ones they're mining if you placed it even at their side as they dug one block ahead they'd never see those diamonds as they consider that area mined out.
As I understand it: 
![2019-05-31_18 59 03](https://user-images.githubusercontent.com/5202568/58725080-34960e00-83d6-11e9-80e6-0d443a83b0d9.png)
When the top block ahead of the miner is broken they can vein mine as far as they can see in the blue area, the bottom block vein mine as far as they can see in the green and the red is ignored. This might be why there's bug #197 and #129

**Comment by Kylerules08** [comment]
same with me 
![2019-05-31_19 38 52](https://user-images.githubusercontent.com/51084028/58740128-9ce7ee00-83dc-11e9-9609-3eadde21cfb9.png)

**Comment by Galarn20** [comment]
linking to https://github.com/TangoTek/TekTopia-Community/issues/197

**Comment by JohnyDL** [comment]
Yeah the more I see the more I think this is a case of the miner getting distracted between mining the block and doing the vein mining, Magleose seems double odd though if that's the case.

A solution is simply fill in the blocks of the tunnel where the ore is and see if the miner mines those blogs then the vein.

**Comment by me2365** [comment]
This happened to me with gold

**Comment by KimHartley24** [comment]
I am having this same issue. I am using a customized world with all the ores spawning between 50-55 height so it's really easy to see how the miner is skipping over the entire (slice?). One block wide, the floor, ceiling, and walls are all ore that he's just walked past. And it's more frequent than maybe would be thought... the miner doesn't scan a "slice" of blocks every few steps... The start of a mineshaft usually goes pretty smooth and then about 15-20 blocks in he starts   skipping slices

**Comment by JJSax** [comment]
This is due to how they generate ores.  They don't actually mine the blocks there.  It randomly chooses an ore when the villager mines stone.  You can test this by making a huge cube of just smooth stone with commands, even high up in the sky.  Then set your miners to mine down the stone shafts.  They will still manage to find ores like iron, gold, and even diamonds despite a high y level.

**Comment by KimHartley24** [comment]
JJSax, these smooth-stone ores are 'bonus' ores, meant to make miners more worth using. They -do- mine ore directly from the ore deposit blocks as a primary function.

**Comment by JJSax** [comment]
Kimhartley24, oh interesting. I did not know this.  Then in that case I ran into this too.  That was the whole reason I had tested in the first place.

**Comment by MorsDux** [comment]
Happens with every ore. If you place blocks where the ore is left the miner finds the left ore.
